src/main.rs

```diff
@@ -34,12 +34,17 @@ use std::sync::Arc;
 use std::time;
 
 use atty::Stream;
-use regex::{RegexBuilder, RegexSetBuilder};
+use regex::{RegexBuilder, RegexSetBuilder, Regex};
 
 use exec::CommandTemplate;
-use internal::{error, pattern_has_uppercase_char, transform_args_with_exec, FdOptions, FileTypes};
+use internal::{error, pattern_has_uppercase_char, transform_args_with_exec, FdOptions, FileTypes,
+               SizeFilter};
 use lscolors::LsColors;
 
+lazy_static! {
+    static ref VALIDATE_SIZE: Regex = { Regex::new(r"^[\+-]{1}\d+[bBkKmMgGTt]{1,2}$").unwrap() };
```

Comment:
I'm going to review this regex piece by piece :smile: 

- `[\+-]{1}` - unneccesary `{1}`?

- `\d+` - Just an idea... do we want to allow for floating point numbers as well?

- `[bBkKmMgGTt]{1,2}` - This allows for a lot of wrong patterns like `bb`, `mm` or `bk`. I think we should separate the unit-part from the optional `byte` annotation and write something like `[kKmMgGtT]?[bB]?`.

Also, see my other comment regarding the units.

Reply:
- Yes, the `{1}` is not required to qualify this match. 
- I do not believe that adding floating point support would increase functionality at all
- This is true, I will switch it to your suggested pattern.